src/kubernetes.cc

```diff
@@ -796,6 +796,15 @@ void KubernetesReader::WatchMaster(
   if (verbose) {
     LOG(INFO) << "WatchMaster(" << name << "): Contacting " << endpoint;
   }
+  std::mutex last_received_mutex;
+  auto last_received = std::chrono::high_resolution_clock::now();
```

Comment:
Your timeout is measured in integral seconds. You probably don't need `high_resolution_clock`, especially because it may be pointing to `system_clock`, which can be confused by DST changes and such. Why not use `std::steady_clock` instead?